test/e2e/metrics_runtime_input_resources_test.go

```diff
@@ -89,19 +92,32 @@ var _ = Describe(suite.ID(), Label(suite.LabelMetrics), Ordered, func() {
 				resp, err := proxyClient.Get(backendOnlyContainerMetricsEnabledURL)
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
-				g.Expect(resp).To(HaveHTTPBody(
-					WithFlatMetrics(ContainElement(HaveField("Name", BeElementOf(kubeletstats.ContainerMetricsNames)))),
-				))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				containerMetricNames := slices.Concat(kubeletstats.ContainerMetricsNames, k8scluster.ContainerMetricsNames)
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(containerMetricNames))), "Found container metrics in backend that are not part of k8scluster or kubeletstats")
```

Comment:
Since we have a concrete list of expected metrics now, can you try to use the built-in `ConsistOf` matcher instead of the custom `OnlyContainElementsOf`?

Reply:
I had to introduce `OnltContainerElementsOf` as expected slice does not need to have all the metrics from kubeletstats + k8sCluster. it checks it should not have any other metrics apart from the listed in the expected list. Anf Consists of expects `expected slice =result slice` (the order can change)